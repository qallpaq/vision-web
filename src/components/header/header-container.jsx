import React from 'react'
import {isNotAuth} from "../../redux/actions"
import {connect} from "react-redux"
import {Header} from "./header"
import {resIsAuth} from "../../redux/reselect"
import {locStorage} from "../../helpers"


const HeaderContainer = ({isAuth, isNotAuth}) => {

    const removeUser = () => {
        isNotAuth()
        locStorage()
    }

    return <Header removeUser={removeUser}
                   isAuth={isAuth}/>
}

const mapStateToProps = state => ({isAuth: resIsAuth(state)})

export default connect(mapStateToProps, {isNotAuth})(HeaderContainer)
