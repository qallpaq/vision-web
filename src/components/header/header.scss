@import "src/scss/constants";

.header {
  background-color: #2C2C2C;
  border-top: 5px solid #A30000;
  width: 100%;
  height: $header-height;
  position: fixed;
  top: 0;
  z-index: 2;

  &__inner {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 100%;
    width: 100%;
  }
}

.logo {
  text-transform: uppercase;
  font-size: 25px;
  font-style: italic;
  font-weight: bold;
  color: #ffffff;

  &_login {
    color: #333;
    margin-bottom: 20px;
  }
}

.links {
  display: flex;
  justify-content: flex-end;
  width: 50%;
}

.links__item:nth-child(1) {
  margin-right: 20px;
}
