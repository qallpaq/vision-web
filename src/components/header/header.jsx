import React from 'react'
import HeaderButton from "./header-button"
import './header.scss'


export const Header = ({isAuth, removeUser}) => {
    return (
        <header className='header'>
            <div className='container'>
                <div className="header__inner">
                    <span className='logo'>Vision Web</span>
                    <ul className='links'>
                        <li className='links__item'>
                            {isAuth
                                ? <HeaderButton text={'Log Out'} link={'/login'} fn={removeUser}/>
                                : <HeaderButton text={'Log In'} link={'/login'}/>
                            }
                        </li>
                        <li className='links__item'>
                            <HeaderButton text={'Sign Up'} link={'/signup'}/>
                        </li>
                    </ul>
                </div>
            </div>
        </header>
    )
}
