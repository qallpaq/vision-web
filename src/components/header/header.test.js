import React from 'react'
import {shallow} from "enzyme"
import {Header} from "./header"


describe('Header', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallow(<Header/>)
    })

    it('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should contain span.logo', () => {
        const result = wrapper.find('span.logo').text()
        expect(result).toBe('Vision Web')
    })

    it('should contain 2 .links__item', () => {
        const result = wrapper.find('.links__item')
        expect(result.length).toBe(2)
    })
})
