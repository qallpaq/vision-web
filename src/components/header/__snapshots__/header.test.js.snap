// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__inner"
    >
      <span
        className="logo"
      >
        Vision Web
      </span>
      <ul
        className="links"
      >
        <li
          className="links__item"
        >
          <HeaderButton
            link="/login"
            text="Log In"
          />
        </li>
        <li
          className="links__item"
        >
          <HeaderButton
            link="/signup"
            text="Sign Up"
          />
        </li>
      </ul>
    </div>
  </div>
</header>
`;
